wkhtmltopdf/utils.py

```diff
@@ -119,52 +119,65 @@ def convert_to_pdf(filename, header_filename=None, footer_filename=None, cmd_opt
         cmd_options['header_html'] = header_filename
     if footer_filename is not None:
         cmd_options['footer_html'] = footer_filename
-    return wkhtmltopdf(pages=[filename], **cmd_options)
+    return wkhtmltopdf(pages=filename_list, **cmd_options)
 
-def render_pdf_from_template(input_template, header_template, footer_template, context, request=None, cmd_options=None):
+def create_rendered_file(template, context, request=None):
+    # Create a temporary file of the rendered template with context.
+    # Return the filename for later conversion to PDF.
     debug = getattr(settings, 'WKHTMLTOPDF_DEBUG', settings.DEBUG)
-    cmd_options = cmd_options if cmd_options else {}
 
-    input_file = header_file = footer_file = None
-    header_filename = footer_filename = None
+    temporary_file = None
 
     try:
-        input_file = render_to_temporary_file(
-            template=input_template,
+        temporary_file = render_to_temporary_file(
+            template=template,
             context=context,
             request=request,
             prefix='wkhtmltopdf', suffix='.html',
             delete=(not debug)
         )
 
-        if header_template:
-            header_file = render_to_temporary_file(
-                template=header_template,
-                context=context,
-                request=request,
-                prefix='wkhtmltopdf', suffix='.html',
-                delete=(not debug)
-            )
-            header_filename = header_file.name
-
-        if footer_template:
-            footer_file = render_to_temporary_file(
-                template=footer_template,
-                context=context,
-                request=request,
-                prefix='wkhtmltopdf', suffix='.html',
-                delete=(not debug)
-            )
-            footer_filename = footer_file.name
-
-        return convert_to_pdf(filename=input_file.name,
-                              header_filename=header_filename,
-                              footer_filename=footer_filename,
-                              cmd_options=cmd_options)
+        return temporary_file.name
     finally:
         # Clean up temporary files
-        for f in filter(None, (input_file, header_file, footer_file)):
-            f.close()
+        if temporary_file is not None:
+            temporary_file.close()
```

Comment:
This looks like it will close the `temporary_file` before the function returns. Since `delete=(not debug)` this will result in the temporary file being deleted before it is used if `debug` is `False` (e.g. in production). 

Not sure if there is a test for this but we should probably be ensuring the tests are run with `DEBUG` off.